src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -242,9 +246,16 @@ export class RunQueryAction extends QueryTaskbarAction {
 			if (runCurrentStatement && selection && this.isCursorPosition(selection)) {
 				editor.input.runQueryStatement(selection);
 			} else {
-				// get the selection again this time with trimming
-				selection = editor.getSelection();
-				editor.input.runQuery(selection);
+				if (editor.input.state.isActualExecutionPlanMode) {
+					this.telemetryService.sendActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.ToggleOnActualQueryExecutionPlan);
```

Comment:
Is this the correct action? We're running the query here, not toggling. Did you mean to put this on the toggle action itself?

Reply:
I was going to put this into the toggle action, but telemetry info on this probably wouldn't be as accurate if users kept clicking on the button without executing any queries.

Reply:
The point would be to indicate how many times people clicked the button though (which is still useful to know). Also collecting how many times people ran a query with the actual plan enabled is a separate data point that should also be collected (same with estimated plan). 

Did you check if we're already collecting info on actual/estimated plan executions?

Reply:
We're already collecting that information. I remember having a conversation with Erin about that a while back ago.

Reply:
Alright, please remove this then. Whether you add an event to the button click is up to you (but I'd encourage it, it's something I see being useful - especially since this is in direct response to a customer ask it'd be good to know how many people are using it)